homeassistant/components/yalexs_ble/config_flow.py

```diff
@@ -0,0 +1,246 @@
+"""Config flow for Yale Access Bluetooth integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from bleak_retry_connector import BleakError, BLEDevice
+import voluptuous as vol
+from yalexs_ble import (
+    AuthError,
+    DisconnectedError,
+    PushLock,
+    ValidatedLockConfig,
+    local_name_is_unique,
+)
+from yalexs_ble.const import YALE_MFR_ID
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.loader import async_get_integration
+
+from .const import CONF_KEY, CONF_LOCAL_NAME, CONF_SLOT, DOMAIN
+from .util import async_get_service_info
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_lock(
+    local_name: str, device: BLEDevice, key: str, slot: int
+) -> None:
+    """Validate a lock."""
+    if len(key) != 32:
+        raise InvalidKeyFormat
+    try:
+        bytes.fromhex(key)
+    except ValueError as ex:
+        raise InvalidKeyFormat from ex
+    if not isinstance(slot, int) or slot < 0 or slot > 255:
+        raise InvalidKeyIndex
+    await PushLock(local_name, device.address, device, key, slot).validate()
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Yale Access Bluetooth."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovery_info: BluetoothServiceInfoBleak | None = None
+        self._discovered_devices: dict[str, BluetoothServiceInfoBleak] = {}
+        self._lock_cfg: ValidatedLockConfig | None = None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        self.context["local_name"] = discovery_info.name
+        self._discovery_info = discovery_info
+        self.context["title_placeholders"] = {
+            "name": discovery_info.name,
+            "local_name": discovery_info.name,
+            "address": self._discovery_info.address,
+        }
+        return await self.async_step_user()
+
+    async def async_step_integration_discovery(
```

Comment:
How is this step triggered?

Reply:
It's going to be triggered from outside the integration when I have a way to feed keys from their cloud service in so users can avoid the need to extract them from their iOS or android device

The yale access cloud and august cloud both have the keys in many cases.  Once I add support for the older august locks as well, I plan on having the august integration discover the keys and feed them via integration discovery